Failure cluster [3aba724e...] TestStaticPodExclusion flake

### Failure cluster [3aba724e4e525dc8072c](https://go.k8s.io/triage#3aba724e4e525dc8072c)

##### Error text:
```
=== RUN   TestStaticPodExclusion
    pod_workers_test.go:865: unexpected status for pod 5: &kubelet.podSyncStatus{ctx:context.Context(nil), cancelFn:(context.CancelFunc)(nil), working:false, fullname:"pod1_test1", syncedAt:time.Time{wall:0xc0915814ce3f8126, ext:16716501782, loc:(*time.Location)(0x89d1780)}, terminatingAt:time.Time{wall:0xc0915814ce461284, ext:16716932212, loc:(*time.Location)(0x89d1780)}, startedTerminating:false, deleted:false, gracePeriod:1, evicted:false, terminatedAt:time.Time{wall:0xc0915814ce4c7321, ext:16717350160, loc:(*time.Location)(0x89d1780)}, finished:true, restartRequested:false, notifyPostTerminating:[]chan<- struct {}(nil), statusPostTerminating:[]kubelet.PodStatusFunc(nil)}
--- FAIL: TestStaticPodExclusion (0.39s)

```
#### Recent failures:
[2022/4/24 13:53:44 ci-kubernetes-unit](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-unit/1518105418134458368)

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/109736/pull-kubernetes-unit/1522458909191704576

/kind failing-test
<!-- If this is a flake, please add: /kind flake -->
/kind flake

the test case was add in #107685 
/cc smarterclayton

<!-- Please assign a SIG using: /sig SIG-NAME -->

/sig node

--- a/pkg/kubelet/pod_workers_test.go
+++ b/pkg/kubelet/pod_workers_test.go
@@ -848,6 +848,8 @@ func TestStaticPodExclusion(t *testing.T) {
 		Pod:        newNamedPod("5-static", "test1", "pod1", true),
 		UpdateType: kubetypes.SyncPodUpdate,
 	})
+	// Wait for the previous work to be delivered to the worker
+	drainAllWorkers(podWorkers)
 	channels.Channel("5-static").Hold()
 	podWorkers.UpdatePod(UpdatePodOptions{
 		Pod:        newNamedPod("5-static", "test1", "pod1", true),

